specific/cmo_sale/models/sale.py

```diff
@@ -93,11 +141,41 @@ def _compute_project_number(self):
 
     @api.model
     def _default_covenant(self):
-        convenants = self.env['sale.convenant.description'].search([
-            ['active', '=', True],
-        ])
-        if convenants:
-            return convenants[0].description
+        Description = self.env['sale.covenant.description']
+        covenants = Description.search([('active', '=', True), ])
+        return covenants and covenants[0].description or False
+
+    @api.multi
+    @api.constrains('order_line')
+    def _constrains_order_line(self):
+        self.ensure_one()
+        if not self.order_line:
+            raise ValidationError("Must have at least 1 order line!")
```

Comment:
All message should have translation function, i.e.,
`ValidationError(_('Must have...'))`
Please correct for all your code. !